cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -509,26 +542,28 @@ func (c *machineController) nodeGroupForNode(node *corev1.Node) (*nodegroup, err
 		return nil, nil
 	}
 
-	if ref := machineSetMachineDeploymentRef(machineSet); ref != nil {
-		key := fmt.Sprintf("%s/%s", machineSet.Namespace, ref.Name)
-		machineDeployment, err := c.findMachineDeployment(key)
-		if err != nil {
-			return nil, fmt.Errorf("unknown MachineDeployment %q: %v", key, err)
-		}
-		if machineDeployment == nil {
-			return nil, fmt.Errorf("unknown MachineDeployment %q", key)
-		}
-		nodegroup, err := newNodegroupFromMachineDeployment(c, machineDeployment)
-		if err != nil {
-			return nil, fmt.Errorf("failed to build nodegroup for node %q: %v", node.Name, err)
-		}
-		// We don't scale from 0 so nodes must belong
-		// to a nodegroup that has a scale size of at
-		// least 1.
-		if nodegroup.MaxSize()-nodegroup.MinSize() < 1 {
-			return nil, nil
+	if c.machineDeploymentResource != nil {
+		if ref := machineSetMachineDeploymentRef(machineSet); ref != nil {
+			key := fmt.Sprintf("%s/%s", machineSet.Namespace, ref.Name)
+			machineDeployment, err := c.findMachineDeployment(key)
+			if err != nil {
+				return nil, fmt.Errorf("unknown MachineDeployment %q: %v", key, err)
```

Comment:
should we differentiate this error text from the following clause?

maybe like `"Error finding MachineDeployment %q: %v"`

Reply:
thinking about this more, including the error text in this response is probably enough to differentiate with the other error. i just wanted to make sure that an operator who sees this error had a chance to figure out the root cause, but i think they will.